### PR TITLE
Add prompt for quit buffer.

### PR DESCRIPTION
**Pull request by wsdjeg**
fix_quit

### CODE REVIEW

File: autoload/SpaceVim/mapping.vim [hunk 1]
```diff
@@ -143,7 +143,23 @@ endfunction
 function! SpaceVim#mapping#close_current_buffer() abort
   let buffers = get(g:, '_spacevim_list_buffers', [])
   let bn = bufnr('%')
-  let index = index(buffers, bn) 
+  if getbufvar(bn, '&modified', 0)
+    redraw!
+    echohl WarningMsg
+    echon 'current buffer contains unsaved modification, save file Y/N?'
```
Comment (icearith) [inline autoload/SpaceVim/mapping.vim]: what will happen if a buffer is not opened from a file?

Comment (wsdjeg) [reply autoload/SpaceVim/mapping.vim]: It only check if the buffer is modified，even if it is not opened from  a file.

Comment (icearith) [reply autoload/SpaceVim/mapping.vim]: yap, I am wondering if the file is not opened from a file, how to save it? does the user need to manually input a file name?

**Comment by tracyone** [comment]
You just need to 

```

set confirm

```

**Comment by wsdjeg** [comment]
this is enabled by default，but when press key binding, the error is ugly.

you can see the pic in issue

**Comment by tracyone** [comment]
Weird.

**Comment by tracyone** [comment]
With following minimal config,

```vim
 set nocompatible
set confirm
colorscheme desert "default setting 
filetype plugin indent on
syntax on

```

screen shot 2018-02-04 at 11 21 32

**Comment by wsdjeg** [comment]
But I am not run `:q`, I am calling a func, in this func will will use `:bd`, this will print the function name when get errors. and I do not want to print the function name.

BTW, I hate two line message, it will push my screen up.

**Review by icearith — APPROVED** [review]
I feel like this PR can be merged into master.

**Pull request merged**